src/dodal/common/beamlines/device_factory.py

```diff
@@ -0,0 +1,112 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic, ParamSpec, TypeVar
+
+from bluesky.run_engine import call_in_bluesky_event_loop
+from ophyd_async.core import DEFAULT_TIMEOUT, Device
+
+from dodal.common.beamlines.beamline_utils import ACTIVE_DEVICES
+
+P = ParamSpec("P")
+T = TypeVar("T", bound=Device)
+_skip = bool | Callable[[], bool]
+
+
+@dataclass
+class DeviceInitializationConfig:
+    """
+    eager_connect: connect or raise Exception at startup
+    use_factory_name: use factory name as name of device
+    timeout: timeout for connecting to the device
+    mock: use Signals with mock backends for device
+    skip: mark the factory to be (conditionally) skipped
```

Comment:
```suggestion
    skip: mark the factory to be (conditionally) skipped when beamline is imported by external program
```

Or something to explain it's not relevant when calling ixx.device()

Reply:
done